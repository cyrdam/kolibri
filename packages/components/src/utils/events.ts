// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
