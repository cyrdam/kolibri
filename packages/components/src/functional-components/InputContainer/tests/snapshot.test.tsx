import type { VNode } from '@stencil/core';
import { h } from '@stencil/core';
import { renderFunctionalComponentToSpecPage } from '../../../utils/testing';
import KolInputContainerFc from '../InputContainer';
import KolIconButtonFc from '../../IconButton';

describe('KolInputContainerFc', () => {
	it('should render correctly', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolInputContainerFc />);
		expect(page.root).toMatchSnapshot();
		expect(page.root?.tagName).toBe('DIV');
	});

	it('should render with startAdornment', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolInputContainerFc startAdornment={(<KolIconButtonFc componentName="icon" icon="start-icon" />) as VNode} />
		));
		expect(page.root).toMatchSnapshot();
		expect(page.root?.querySelector('kol-icon')).not.toBeNull();
	});

	it('should render with endAdornment', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolInputContainerFc endAdornment={(<KolIconButtonFc componentName="icon" icon="end-icon" />) as VNode} />
		));
		expect(page.root).toMatchSnapshot();
		expect(page.root?.querySelector('kol-icon')).not.toBeNull();
	});

	it('should render with both startAdornment and endAdornment', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolInputContainerFc
				startAdornment={(<KolIconButtonFc componentName="icon" icon="start-icon" />) as VNode}
				endAdornment={(<KolIconButtonFc componentName="icon" icon="end-icon" />) as VNode}
			/>
		));
		expect(page.root).toMatchSnapshot();
		expect(page.root?.querySelectorAll('kol-icon')).toHaveLength(2);
	});

	it('should render with children', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolInputContainerFc>
				<input type="text" />
			</KolInputContainerFc>
		));
		expect(page.root).toMatchSnapshot();
		expect(page.root?.querySelector('input')).not.toBeNull();
	});
});
