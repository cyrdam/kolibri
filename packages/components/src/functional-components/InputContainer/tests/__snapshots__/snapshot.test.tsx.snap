// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KolInputContainerFc should render correctly 1`] = `<div class="input"></div>`;

exports[`KolInputContainerFc should render with both startAdornment and endAdornment 1`] = `
<div class="input">
  <div class="inputAdornment-start">
    <kol-icon _icons="codicon codicon-start-icon" _label=""></kol-icon>
  </div>
  <div class="inputAdornment-end">
    <kol-icon _icons="codicon codicon-end-icon" _label=""></kol-icon>
  </div>
</div>
`;

exports[`KolInputContainerFc should render with children 1`] = `
<div class="input">
  <input type="text">
</div>
`;

exports[`KolInputContainerFc should render with endAdornment 1`] = `
<div class="input">
  <div class="inputAdornment-end">
    <kol-icon _icons="codicon codicon-end-icon" _label=""></kol-icon>
  </div>
</div>
`;

exports[`KolInputContainerFc should render with startAdornment 1`] = `
<div class="input">
  <div class="inputAdornment-start">
    <kol-icon _icons="codicon codicon-start-icon" _label=""></kol-icon>
  </div>
</div>
`;
